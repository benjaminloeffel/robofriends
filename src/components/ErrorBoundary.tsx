import React, { Component } from "react";

type Props = {};

type State = {
  hasError: boolean;
};

class ErrorBoundary extends Component<Props, State> {
  readonly state: State = {
    hasError: false
  };

  componentDidCatch(error: any, info: any) {
    this.setState({ hasError: true });
  }

  render() {
    if (this.state.hasError) {
      return (
        <div>
          <h1>Oops! Something went wrong!</h1>
        </div>
      );
    } else {
      return this.props.children;
    }
  }
}

export default ErrorBoundary;
