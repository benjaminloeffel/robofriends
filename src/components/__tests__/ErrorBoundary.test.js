import React from "react";
import "../../setupTests";
import { shallow, mount } from "enzyme";
import ErrorBoundary from "../ErrorBoundary";

describe("ErrorBoundary", () => {
  it("renders the ErrorBoundary component", () => {
    const wrapper = shallow(<ErrorBoundary />);
    expect(wrapper).toMatchSnapshot();
  });

  const Something = () => null;
  it("should display an error message if wrapped component throws an error", () => {
    const wrapper = mount(
      <ErrorBoundary>
        <Something />
      </ErrorBoundary>
    );

    const error = new Error("test");
    wrapper.find(Something).simulateError(error);
  });
});
