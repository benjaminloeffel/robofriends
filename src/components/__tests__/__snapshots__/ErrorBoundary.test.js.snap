// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary renders the ErrorBoundary component 1`] = `ShallowWrapper {}`;
